File: C_Sharp_code/Example/WindowsFormsApp/CS_demo20/Apple.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CS_demo20
{
    class Apple
    {
        string _name;
        public Apple(string argName)
        {
            _name = argName;
        }

        public override bool Equals(object b)
        {
            Apple a;
            a = (Apple)b;//將b轉型存到變數a
            if (this._name.Equals(a._name))
                return true;
            else
                return false;
        }

        public override string ToString()
        {
            return "Name: " + _name;
        }

    }
}

File: C_Sharp_code/Example/WindowsFormsApp/VB_demo21/Apple.vb
Public Class Apple
    Public _name As String
    Public Sub New(ByVal argName As String)
        _name = argName
    End Sub

    Public Overrides Function ToString() As String
        Return "Name: " & _name
    End Function

    Public Overrides Function Equals(obj As Object) As Boolean
        Dim x As Apple
        x = CType(obj, Apple) '將obj轉型存到變數x
        If (Me._name.Equals(x._name)) Then
            Return True
        Else
            Return False
        End If
    End Function
End Class
